stdlib/universe/join.go

```diff
@@ -145,6 +145,15 @@ func createJoinOpSpec(args flux.Arguments, a *flux.Administration) (flux.Operati
 }
 
 func (t *JoinOpSpec) IDer(ider flux.IDer) {
+	// When join is used inside a `tableFind`, the input table operations are
+	// "cloned" granting them new ids.
+	// This will cause a panic to occur in `createMergeJoinTransformation` when
+	// trying to associate table names with their related datasets if we don't
+	// first remove old entries from the map.
+	// Refs: <https://github.com/influxdata/flux/issues/4692>
+	for k := range t.TableNames {
+		delete(t.TableNames, k)
+	}
```

Comment:
My general thinking here is internal to the OpSpec, we rely on `param` field more than the public `TableNames` field so if `param` is changing, we should "fit" `TableNames` to it.

This solves the panic problem by keeping the two aligned, so to speak.